Sources/Core/EthereumABI/ABIElements.swift

```diff
@@ -325,7 +325,7 @@ extension ABI.Element.Function {
             // set a flag to detect the request succeeded
         }
 
-        if returnArray.isEmpty {
+        if returnArray.isEmpty && !outputs.isEmpty {
```

Comment:
While trying to understand what is the issue this line change fixes I understood that we do not have clear explanation for this function outputs.
It's not clear if this change won't just break someone's current use of this library. It's a weird edge case but someone could just be checking if returned data `!= nil`. Thus, we shall clearly specify what to expect in each return case.
What I see so far with current change:
- `nil` is returned if something has failed (big minus to us that we don't tell yet what the issue was; must be improved);
- `["_success": true]` for cases when `ABI.Element.Function` has no return types specified;
- `["_success": true, ....]` for cases when some return types have been specified and data successfully decoded.

Reply:
pharms-eth gave an example here: https://github.com/web3swift-team/web3swift/pull/670#discussion_r1026389333

Reply:
I've almost prepared documentation for this function but had to unpack each path leading to `return` in this function.
It looks like there could be some issues in the function itself when it comes to decoding the invocation of `require` statement in https://github.com/web3swift-team/web3swift/pull/685/files#diff-3c274f0e8aa4f7a52ab734f9d55c13954e03532655ed01e4ad51747d03fb10e4R304-R312

I want to thoroughly debug and cover it with tests.